### PR TITLE
Fix get_dates function

### LINKED ISSUE #437: get_dates issues with date objects

### PR DESCRIPTION
**Pull request by mwinitch**
Closes #437

### CODE REVIEW

**Comment by mwinitch** [comment]
@sbillinge This is ready for review. The get_dates function can have issues because when you do `thing.get('begin_date)` it will sometimes return a datetime object instead of a string. However, especially when it is getting dates from the schemas, it does return a valid string. This fixes the function so it works regardless if `.get` returns a datetime object or a string that has to be converted to a datetime object.

**Comment by mwinitch** [comment]
Yes I added some tests for it. Just wondering how come this needs news but the two other fixes I did for l_projecta did not news?

**Comment by mwinitch** [comment]
@sbillinge This is ready for review

File: news/str_date.rst [hunk 1]
```diff
@@ -0,0 +1,12 @@
+**Added:**
+    * get_dates function that now works with objects with datetime objects
```
Comment (sbillinge) [inline news/str_date.rst:2]: I think you didn't add a get_dates function, you fixed it.... (see below)

Comment (mwinitch) [reply news/str_date.rst]: Fixed

**Comment by mwinitch** [comment]
@sbillinge News is fixed and it is ready for merge

**Comment by sbillinge** [comment]
Thanks Michael, this is really awesome!

**Pull request merged**